src/frontend/lib/utils.js

```diff
@@ -142,6 +143,35 @@ export function formatCoords({
   }
 }
 
+export function getProp(tags: any, fieldKey: Key, defaultValue: any) {
+  // TODO: support deeply nested tags.
+  const shallowKey = Array.isArray(fieldKey) ? fieldKey[0] : fieldKey;
+  const tagValue = tags[shallowKey];
+  return typeof tagValue === "undefined" ? defaultValue : tagValue;
+}
+
+/**
+ * Convert a select option which could either be a string or an object with
+ * label and value props, to an object with label and value props TODO: Show
+ * meaningful translated values for null and boolean, but these types are not
+ * used widely in presets yet
+ */
+export function convertSelectOptionsToLabeled(
```

Comment:
This also seems like it will be important to re-use this same rendering logic in other UIs, but it is not a react component. If the only argument is a SelectOptions object, I wonder how we can make it more tightly coupled, so it isn't in a 'utils' file?

Reply:
Yeah I have been wondering about that. This is used by:

1. Select One data editing component
2. Select Multiple data editing component
3. Select one and multiple data view component

On desktop the editing component will likely be different (due to different UI patterns on Desktop) but will also need this function. As such I think it needs to be a helper function shared by these components. I think we can drop support for this in a later version of Mapeo Presets, or perhaps do this transform on the server. This is necessary because we originally supported options to be an array of strings, but later supported labels and values. Once we have a preset editor, these simplicity of just adding an array of strings is no longer so important.